common/djangoapps/student/models.py

```diff
@@ -284,6 +279,46 @@ def set_login_session(self, session_id=None):
         self.set_meta(meta)
         self.save()
 
+    def requires_parental_consent(self, date=None, age_limit=None, default_requires_consent=True):
+        """Returns true if this user requires parental consent.
+
+        Args:
+            date (Date): The date for which consent needs to be tested (defaults to now).
+            age_limit (int): The age limit at which parental consent is no longer required.
+                This defaults to the value of the setting 'PARENTAL_CONTROL_AGE_LIMIT'.
+            default_requires_consent (Bool): True if users require parental consent if they
+                have no specified year of birth (default is True).
+
+        Returns:
+             True if the user requires parental consent.
+        """
+        if age_limit is None:
+            age_limit = getattr(settings, 'PARENTAL_CONSENT_AGE_LIMIT', None)
+            if age_limit is None:
+                return False
+
+        # Return True if either:
+        # a) The user has a year of birth specified and that year is fewer years in the past than the limit.
+        # b) The user has no year of birth specified and the default is to require consent.
+        year_of_birth = self.year_of_birth
+        if year_of_birth is None:
+            return default_requires_consent
+        if date is None:
+            date = datetime.now(UTC)
+        return date.year - year_of_birth <=  age_limit # pylint: disable=maybe-no-member
```

Comment:
<= limit to be conservative enough in case the birthdate month is later in the year than the current month?

I think there is an extra space after <=, btw.